[DataGridPremium] Undo and redo (#20594)
mui/mui-x: docs/data/data-grid/undo-redo/undo-redo.md

<p class="description">Let users undo and reapply changes made to the Data Grid.</p>

The undo/redo feature lets users reverse and reapply their actions in the Data Grid.
It tracks registered events, listens for keyboard events and adds toolbar buttons.

## Basic usage

The undo/redo feature is enabled when at least one event handler is registered and `historyQueueSize` is greater than 0.
The toolbar automatically displays undo and redo buttons when the feature is active.

Users can:

- Undo an action using <kbd><kbd class="key">Ctrl</kbd>+<kbd class="key">Z</kbd></kbd> (<kbd><kbd class="key">⌘ Command</kbd>+<kbd class="key">Z</kbd></kbd> on macOS)
- Redo an action using <kbd><kbd class="key">Ctrl</kbd>+<kbd class="key">Shift</kbd>+<kbd class="key">Z</kbd></kbd> and <kbd><kbd class="key">Ctrl</kbd>+<kbd class="key">Y</kbd></kbd> (<kbd><kbd class="key">⌘ Command</kbd>+<kbd class="key">Shift</kbd>+<kbd class="key">Z</kbd></kbd> and <kbd><kbd class="key">⌘ Command</kbd>+<kbd class="key">Y</kbd></kbd> on macOS)

{{"demo": "BasicUndoRedo.js", "bg": "inline", "defaultCodeOpen": false}}

### Disabling the undo and redo

> To disable the undo/redo feature, set `historyQueueSize` to `0`:

Reviewer: A small demo for this with a toggle button would be useful.

Author: Added

```tsx
<DataGridPremium historyQueueSize={0} />
```

This will prevent any history from being tracked and hide the undo/redo buttons from the toolbar.

### Removing the toolbar buttons

> To only remove the toolbar buttons for undo and redo, but keep the keyboard controls active, set the `showUndoRedo` flag to `false` in the Toolbar's slot props.

Reviewer 2: ```suggestion
To remove the toolbar buttons for undo and redo while keeping the keyboard controls active, set the `showUndoRedo` flag to `false` in the Toolbar's slot props.
```
